File: ncPersistencia/pClienteFinanceiro.vb
Imports ncDados.nsCliente
Imports ncComum.nsAcessoBD
Imports ncComum.nsFuncoes
Imports ncComum.nsExcecao

Namespace nsCliente

  Public Class pClienteFinanceiro

    Public Function Listar() As ColecaoClienteFinanceiro

      Dim retorno As ColecaoClienteFinanceiro
      Dim acessoBanco As cAcessoBD
      Dim ds As DataSet
      Dim dt As DataTable
      Dim row As DataRow
      Dim item As dClienteFinanceiro
      Dim comandoSQL As String

      Try

        acessoBanco = New cAcessoBD

        comandoSQL = " Select limite, situacaoCrediario, cliente_cid, " & _
                      "banco1, banco2, agencia1, agencia2, conta1, conta2, gerente1, gerente2, referencia1, referencia2, ddd1, ddd2, telefone1, telefone2, observacoes " & _
                      " From clientefinanceiro "

        ds = acessoBanco.ExecutarDS(comandoSQL)

        If Not ds Is Nothing Then
          If ds.Tables.Count > 0 Then
            dt = ds.Tables(0)

            If dt.Rows.Count > 0 Then
              retorno = New ColecaoClienteFinanceiro

              For Each row In dt.Rows
                item = New dClienteFinanceiro

                item.limite = cFuncoes.RetornarDecimal(row("limite"))
                item.situacaoCrediario = cFuncoes.RetornarTexto(row("situacaoCrediario"))
                item.cliente_cid = cFuncoes.RetornarInteiro(row("cliente_cid"))
                item.banco1 = cFuncoes.RetornarTexto(row("banco1"))
                item.banco2 = cFuncoes.RetornarTexto(row("banco2"))
                item.agencia1 = cFuncoes.RetornarTexto(row("agencia1"))
                item.agencia2 = cFuncoes.RetornarTexto(row("agencia2"))
                item.conta1 = cFuncoes.RetornarTexto(row("conta1"))
                item.conta2 = cFuncoes.RetornarTexto(row("conta2"))
                item.gerente1 = cFuncoes.RetornarTexto(row("gerente1"))
                item.gerente2 = cFuncoes.RetornarTexto(row("gerente2"))
                item.referencia1 = cFuncoes.RetornarTexto(row("referencia1"))
                item.referencia2 = cFuncoes.RetornarTexto(row("referencia2"))
                item.telefone1 = cFuncoes.RetornarTexto(row("telefone1"))
                item.telefone2 = cFuncoes.RetornarTexto(row("telefone2"))
                item.ddd1 = cFuncoes.RetornarTexto(row("ddd1"))
                item.ddd2 = cFuncoes.RetornarTexto(row("ddd2"))
                item.observacoes = cFuncoes.RetornarTexto(row("observacoes"))

                retorno.Add(item)
              Next
            Else
              retorno = Nothing
            End If
          Else
            retorno = Nothing
          End If
        Else
          retorno = Nothing
        End If

      Catch ex As Exception

        retorno = Nothing
        Throw New ExcecaoNascomercio("Erro em Listar Cliente - Financeiro [" & Me.ToString() & "] - " & ex.Message)

      End Try

      Listar = retorno

    End Function

    Public Function Consultar(ByVal dados As dClienteFinanceiro) As ColecaoClienteFinanceiro

      Dim retorno As ColecaoClienteFinanceiro
      Dim acessoBanco As cAcessoBD
      Dim ds As DataSet
      Dim dt As DataTable
      Dim row As DataRow
      Dim item As dClienteFinanceiro
      Dim sqlSelect As String
      Dim sqlWhere As String
      Dim sqlFrom As String

      Try

        acessoBanco = New cAcessoBD

        sqlSelect = " Select limite, situacaoCrediario, cliente_cid, " & _
                      "banco1, banco2, agencia1, agencia2, conta1, conta2, gerente1, gerente2, referencia1, referencia2, ddd1, ddd2, telefone1, telefone2, observacoes "
        sqlWhere = String.Empty
        sqlFrom = " From clientefinanceiro "

        '-- limite
        sqlWhere = cFuncoes.MontarParametrosSQL(sqlWhere, dados.limite, "limite")

        '-- situacaoCredito
        sqlWhere = cFuncoes.MontarParametrosSQL(sqlWhere, dados.situacaoCrediario, "situacaoCrediario")

        '-- cliente_cid
        sqlWhere = cFuncoes.MontarParametrosSQL(sqlWhere, dados.cliente_cid, "cliente_cid")

        If Not sqlWhere.Equals(String.Empty) Then
          sqlWhere = " WHERE " & sqlWhere
        End If

        ds = acessoBanco.ExecutarDS(sqlSelect & " " & sqlFrom & " " & sqlWhere)

        If Not ds Is Nothing Then
          If ds.Tables.Count > 0 Then
            dt = ds.Tables(0)

            If dt.Rows.Count > 0 Then
              retorno = New ColecaoClienteFinanceiro

              For Each row In dt.Rows
                item = New dClienteFinanceiro

                item.limite = cFuncoes.RetornarDecimal(row("limite"))
                item.situacaoCrediario = cFuncoes.RetornarTexto(row("situacaoCrediario"))
                item.cliente_cid = cFuncoes.RetornarInteiro(row("cliente_cid"))
                item.banco1 = cFuncoes.RetornarTexto(row("banco1"))
                item.banco2 = cFuncoes.RetornarTexto(row("banco2"))
                item.agencia1 = cFuncoes.RetornarTexto(row("agencia1"))
                item.agencia2 = cFuncoes.RetornarTexto(row("agencia2"))
                item.conta1 = cFuncoes.RetornarTexto(row("conta1"))
                item.conta2 = cFuncoes.RetornarTexto(row("conta2"))
                item.gerente1 = cFuncoes.RetornarTexto(row("gerente1"))
                item.gerente2 = cFuncoes.RetornarTexto(row("gerente2"))
                item.referencia1 = cFuncoes.RetornarTexto(row("referencia1"))
                item.referencia2 = cFuncoes.RetornarTexto(row("referencia2"))
                item.telefone1 = cFuncoes.RetornarTexto(row("telefone1"))
                item.telefone2 = cFuncoes.RetornarTexto(row("telefone2"))
                item.ddd1 = cFuncoes.RetornarTexto(row("ddd1"))
                item.ddd2 = cFuncoes.RetornarTexto(row("ddd2"))
                item.observacoes = cFuncoes.RetornarTexto(row("observacoes"))

                retorno.Add(item)
              Next
            Else
              retorno = Nothing
            End If
          Else
            retorno = Nothing
          End If
        Else
          retorno = Nothing
        End If

      Catch ex As Exception

        retorno = Nothing
        Throw New ExcecaoNascomercio("Erro em Consultar Cliente - Financeiro [" & Me.ToString() & "] - " & ex.Message)

      End Try

      Consultar = retorno

    End Function

    Public Function Incluir(ByVal dados As dClienteFinanceiro) As Integer

      Dim retorno As Integer
      Dim acessoBanco As cAcessoBD
      Dim comandoSQL As String

      Try

        acessoBanco = New cAcessoBD

        comandoSQL = " INSERT INTO " & _
            " clientefinanceiro (limite, situacaoCrediario, cliente_cid, " & _
            " banco1, banco2, agencia1, agencia2, conta1, conta2, gerente1, gerente2, referencia1, referencia2, ddd1, ddd2, telefone1, telefone2, observacoes) " & _
            " VALUES (" & _
            cFuncoes.PersistirDecimal(dados.limite) & "," & _
            cFuncoes.PersistirTexto(dados.situacaoCrediario) & "," & _
            cFuncoes.PersistirInteiro(dados.cliente_cid) & "," & _
            cFuncoes.PersistirTexto(dados.banco1) & "," & _
            cFuncoes.PersistirTexto(dados.banco2) & "," & _
            cFuncoes.PersistirTexto(dados.agencia1) & "," & _
            cFuncoes.PersistirTexto(dados.agencia2) & "," & _
            cFuncoes.PersistirTexto(dados.conta1) & "," & _
            cFuncoes.PersistirTexto(dados.conta2) & "," & _
            cFuncoes.PersistirTexto(dados.gerente1) & "," & _
            cFuncoes.PersistirTexto(dados.gerente2) & "," & _
            cFuncoes.PersistirTexto(dados.referencia1) & "," & _
            cFuncoes.PersistirTexto(dados.referencia2) & "," & _
            cFuncoes.PersistirTexto(dados.ddd1) & "," & _
            cFuncoes.PersistirTexto(dados.ddd2) & "," & _
            cFuncoes.PersistirTexto(dados.telefone1) & "," & _
            cFuncoes.PersistirTexto(dados.telefone2) & "," & _
            cFuncoes.PersistirTexto(dados.observacoes) & ")"

        retorno = acessoBanco.ExecutarCID(comandoSQL)

      Catch ex As Exception

        retorno = Nothing
        Throw New ExcecaoNascomercio("Erro em Incluir Cliente - Financeiro [" & Me.ToString() & "] - " & ex.Message)

      End Try

      Incluir = retorno

    End Function

    Public Function Alterar(ByVal dados As dClienteFinanceiro) As Integer

      Dim retorno As Integer
      Dim acessoBanco As cAcessoBD
      Dim comandoSQL As String

      Try

        acessoBanco = New cAcessoBD

        comandoSQL = " UPDATE clientefinanceiro SET " & _
            " limite = " & cFuncoes.PersistirDecimal(dados.limite) & "," & _
            " situacaoCrediario = " & cFuncoes.PersistirTexto(dados.situacaoCrediario) & "," & _
            " banco1 = " & cFuncoes.PersistirTexto(dados.banco1) & "," & _
            " banco2 = " & cFuncoes.PersistirTexto(dados.banco2) & "," & _
            " agencia1 = " & cFuncoes.PersistirTexto(dados.agencia1) & "," & _
            " agencia2 = " & cFuncoes.PersistirTexto(dados.agencia2) & "," & _
            " conta1 = " & cFuncoes.PersistirTexto(dados.conta1) & "," & _
            " conta2 = " & cFuncoes.PersistirTexto(dados.conta2) & "," & _
            " gerente1 = " & cFuncoes.PersistirTexto(dados.gerente1) & "," & _
            " gerente2 = " & cFuncoes.PersistirTexto(dados.gerente2) & "," & _
            " referencia1 = " & cFuncoes.PersistirTexto(dados.referencia1) & "," & _
            " referencia2 = " & cFuncoes.PersistirTexto(dados.referencia2) & "," & _
            " ddd1 = " & cFuncoes.PersistirTexto(dados.ddd1) & "," & _
            " ddd2 = " & cFuncoes.PersistirTexto(dados.ddd2) & "," & _
            " telefone1 = " & cFuncoes.PersistirTexto(dados.telefone1) & "," & _
            " telefone2 = " & cFuncoes.PersistirTexto(dados.telefone2) & "," & _
            " observacoes = " & cFuncoes.PersistirTexto(dados.observacoes) & _
            " WHERE " & _
            " cliente_cid = " & cFuncoes.PersistirInteiro(dados.cliente_cid)

        retorno = acessoBanco.ExecutarINT(comandoSQL)

      Catch ex As Exception

        retorno = Nothing
        Throw New ExcecaoNascomercio("Erro em Alterar Cliente - Financeiro [" & Me.ToString() & "] - " & ex.Message)

      End Try

      Alterar = retorno

    End Function

    Public Function ExcluirPorCliente(ByVal cliente_cid As Integer) As Integer

      Dim retorno As Integer
      Dim acessoBanco As cAcessoBD
      Dim comandoSQL As String

      Try

        acessoBanco = New cAcessoBD

        comandoSQL = " DELETE FROM clientefinanceiro " & _
            " WHERE cliente_cid = " & cliente_cid.ToString()

        retorno = acessoBanco.ExecutarINT(comandoSQL)

      Catch ex As Exception

        retorno = Nothing
        Throw New ExcecaoNascomercio("Erro em Excluir Cliente - Financeiro [" & Me.ToString() & "] - " & ex.Message)

      End Try

      ExcluirPorCliente = retorno

    End Function

  End Class

End Namespace

File: ncRepositorios/pClienteFinanceiro.cs
using System;
using System.Collections.Generic;
using Dapper;
using MySql.Data.MySqlClient;
using ncNComum.nsExcecao;
using nsCliente;

namespace ncPersistencia.nsCliente
{
    public class pClienteFinanceiro : RepositorioBase, IpClienteFinanceiro
    {
        public ColecaoClienteFinanceiro Listar()
        {
            try
            {
                using (var conn = CriarConexao())
                {
                    var lista = conn.Query<dClienteFinanceiro>(
                        "SELECT limite, situacaoCrediario, cliente_cid, " +
                        "banco1, banco2, agencia1, agencia2, conta1, conta2, gerente1, gerente2, referencia1, referencia2, ddd1, ddd2, telefone1, telefone2, observacoes " +
                        "FROM clientefinanceiro").AsList();
                    if (lista.Count == 0) return null;
                    var retorno = new ColecaoClienteFinanceiro();
                    retorno.AddRange(lista);
                    return retorno;
                }
            }
            catch (ExcecaoNascomercio) { throw; }
            catch (Exception ex)
            {
                throw new ExcecaoNascomercio("Erro em Listar Cliente - Financeiro [" + ToString() + "] - " + ex.Message);
            }
        }

        public ColecaoClienteFinanceiro Consultar(dClienteFinanceiro dados)
        {
            try
            {
                using (var conn = CriarConexao())
                {
                    var conditions = new List<string>();
                    var p = new DynamicParameters();
                    if (dados.limite != null && dados.limite != 0) { conditions.Add("limite=@limite"); p.Add("limite", dados.limite); }
                    if (!string.IsNullOrEmpty(dados.situacaoCrediario)) { conditions.Add("situacaoCrediario=@situacaoCrediario"); p.Add("situacaoCrediario", dados.situacaoCrediario); }
                    if (dados.cliente_cid != null && dados.cliente_cid != 0) { conditions.Add("cliente_cid=@cliente_cid"); p.Add("cliente_cid", dados.cliente_cid); }
                    var where = conditions.Count > 0 ? "WHERE " + string.Join(" AND ", conditions) : "";
                    var sql = $"SELECT limite, situacaoCrediario, cliente_cid, " +
                              $"banco1, banco2, agencia1, agencia2, conta1, conta2, gerente1, gerente2, referencia1, referencia2, ddd1, ddd2, telefone1, telefone2, observacoes " +
                              $"FROM clientefinanceiro {where}";
                    var lista = conn.Query<dClienteFinanceiro>(sql, p).AsList();
                    if (lista.Count == 0) return null;
                    var retorno = new ColecaoClienteFinanceiro();
                    retorno.AddRange(lista);
                    return retorno;
                }
            }
            catch (ExcecaoNascomercio) { throw; }
            catch (Exception ex)
            {
                throw new ExcecaoNascomercio("Erro em Consultar Cliente - Financeiro [" + ToString() + "] - " + ex.Message);
            }
        }

        public int Incluir(dClienteFinanceiro dados)
        {
            try
            {
                using (var conn = CriarConexao())
                {
                    conn.Execute(
                        "INSERT INTO clientefinanceiro (limite, situacaoCrediario, cliente_cid, " +
                        "banco1, banco2, agencia1, agencia2, conta1, conta2, gerente1, gerente2, referencia1, referencia2, ddd1, ddd2, telefone1, telefone2, observacoes) " +
                        "VALUES (@limite, @situacaoCrediario, @cliente_cid, " +
                        "@banco1, @banco2, @agencia1, @agencia2, @conta1, @conta2, @gerente1, @gerente2, @referencia1, @referencia2, @ddd1, @ddd2, @telefone1, @telefone2, @observacoes)",
                        new {
                            limite = dados.limite, situacaoCrediario = dados.situacaoCrediario, cliente_cid = dados.cliente_cid,
                            banco1 = dados.banco1, banco2 = dados.banco2, agencia1 = dados.agencia1, agencia2 = dados.agencia2,
                            conta1 = dados.conta1, conta2 = dados.conta2, gerente1 = dados.gerente1, gerente2 = dados.gerente2,
                            referencia1 = dados.referencia1, referencia2 = dados.referencia2, ddd1 = dados.ddd1, ddd2 = dados.ddd2,
                            telefone1 = dados.telefone1, telefone2 = dados.telefone2, observacoes = dados.observacoes
                        });
                    return (int)conn.ExecuteScalar<long>("SELECT LAST_INSERT_ID()");
                }
            }
            catch (ExcecaoNascomercio) { throw; }
            catch (Exception ex)
            {
                throw new ExcecaoNascomercio("Erro em Incluir Cliente - Financeiro [" + ToString() + "] - " + ex.Message);
            }
        }

        public int Alterar(dClienteFinanceiro dados)
        {
            try
            {
                using (var conn = CriarConexao())
                {
                    return conn.Execute(
                        "UPDATE clientefinanceiro SET limite=@limite, situacaoCrediario=@situacaoCrediario, " +
                        "banco1=@banco1, banco2=@banco2, agencia1=@agencia1, agencia2=@agencia2, " +
                        "conta1=@conta1, conta2=@conta2, gerente1=@gerente1, gerente2=@gerente2, " +
                        "referencia1=@referencia1, referencia2=@referencia2, ddd1=@ddd1, ddd2=@ddd2, " +
                        "telefone1=@telefone1, telefone2=@telefone2, observacoes=@observacoes " +
                        "WHERE cliente_cid=@cliente_cid",
                        new {
                            limite = dados.limite, situacaoCrediario = dados.situacaoCrediario,
                            banco1 = dados.banco1, banco2 = dados.banco2, agencia1 = dados.agencia1, agencia2 = dados.agencia2,
                            conta1 = dados.conta1, conta2 = dados.conta2, gerente1 = dados.gerente1, gerente2 = dados.gerente2,
                            referencia1 = dados.referencia1, referencia2 = dados.referencia2, ddd1 = dados.ddd1, ddd2 = dados.ddd2,
                            telefone1 = dados.telefone1, telefone2 = dados.telefone2, observacoes = dados.observacoes,
                            cliente_cid = dados.cliente_cid
                        });
                }
            }
            catch (ExcecaoNascomercio) { throw; }
            catch (Exception ex)
            {
                throw new ExcecaoNascomercio("Erro em Alterar Cliente - Financeiro [" + ToString() + "] - " + ex.Message);
            }
        }

        public int ExcluirPorCliente(int cliente_cid)
        {
            try
            {
                using (var conn = CriarConexao())
                {
                    return conn.Execute("DELETE FROM clientefinanceiro WHERE cliente_cid=@cliente_cid", new { cliente_cid });
                }
            }
            catch (ExcecaoNascomercio) { throw; }
            catch (Exception ex)
            {
                throw new ExcecaoNascomercio("Erro em Excluir Cliente - Financeiro [" + ToString() + "] - " + ex.Message);
            }
        }
    }
}
